app/models/concerns/arclight/solr_document.rb

```diff
@@ -51,19 +41,25 @@ def unitid
     end
 
     def repository
-      first('repository_ssm')
+      first('repository_ssm') || collection&.first('repository_ssm')
     end
 
     def repository_and_unitid
       [repository, unitid].compact.join(': ')
     end
 
+    def collection
+      return self if collection?
+
+      @collection ||= self.class.new(self['collection']&.dig('docs', 0), @response)
+    end
+
     def collection_name
-      first('collection_ssm')
+      collection&.normalized_title
```

Comment:
is there a reason it wouldn't have a collection?

Reply:
Most likely, bad fixture data. There's some indication in the indexing code that the top-level EAD might be something other than a collection, though, so it didn't feel too inappropriate to me 🤷‍♂️